### PR TITLE
Precision option for tensor display

### PR DESCRIPTION
**Pull request by antimora**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Changes

1. Add global precision option
2. Add formatter precision option
3. Add unit tests
4. Improve locking of the global option

### Testing

New unit test for formatter precision option. The global precision settings were tested manually because the change affects other test threads.

### CODE REVIEW

File: crates/burn-tensor/src/tests/stats/display.rs [hunk 1]
```diff
@@ -281,4 +281,55 @@ mod tests {
...
+  backend:  {:?},
+  kind:  "Float",
+  dtype:  "f32",
+}}"#,
+            tensor.device(),
+            TestBackend::name(),
+        );
+        assert_eq!(output, expected);
+
+        // set_print_options(Default::default());
```
Comment (mepatrick73) [inline crates/burn-tensor/src/tests/stats/display.rs:333]: commented code

File: crates/burn-tensor/src/tests/stats/display.rs [hunk 2]
```diff
@@ -281,4 +281,55 @@ mod tests {
...
+  backend:  {:?},
+  kind:  "Float",
+  dtype:  "f32",
+}}"#,
+            tensor.device(),
+            TestBackend::name(),
+        );
+        assert_eq!(output, expected);
+
+        // CAN'T DO THIS BECAUSE OF GLOBAL STATE
```
Comment (mepatrick73) [inline crates/burn-tensor/src/tests/stats/display.rs:304]: commented code

File: burn-book/src/building-blocks/tensor.md [hunk 3]
```diff
@@ -332,3 +332,80 @@ strategies.
...
+
+For more fine-grained control over tensor printing, Burn provides a `PrintOptions` struct and a
+`set_print_options` function:
+
+```rust
+use burn::tensor::{set_print_options, PrintOptions};
+
+let print_options = PrintOptions {
+    precision: Some(2),
+    // threshold: 1000,
```
Comment (mepatrick73) [inline burn-book/src/building-blocks/tensor.md:398]: commented code

**Review by mepatrick73 — APPROVED** [review]
LGTM. Just a few left out comments to be removed. I'll approve in advance :+1:

**Review by laggui — COMMENTED** [review]
Ohh that's pretty useful! Thanks for adding this 😄

**Comment by antimora** [comment]
Thanks for reviewing my precision PR. 

The left comments were intentional. Do you think I should still remove them?

**Review by laggui — APPROVED** [review]
The only comment left to document restrictions seems fine to me :) thanks for the changes

File: crates/burn-tensor/src/tests/stats/display.rs (hunk 1 above)
Comment (antimora) [reply crates/burn-tensor/src/tests/stats/display.rs]: Removed

File: crates/burn-tensor/src/tests/stats/display.rs (hunk 2 above)
Comment (antimora) [reply crates/burn-tensor/src/tests/stats/display.rs:304]: Leaving this one as a note.

File: burn-book/src/building-blocks/tensor.md (hunk 3 above)
Comment (antimora) [reply burn-book/src/building-blocks/tensor.md]: Removed

**Pull request merged**